ui/src/screens/EntityScreen/EntityScreen.jsx

```diff
@@ -53,6 +54,8 @@ class EntityScreen extends Component {
       );
     }
 
+    setRecentlyViewedItem(entityId);
```

Comment:
This maybe wants to live in componentDidMount? so that it's not being called on every render?